proposed/2022/pack-referenced-projects.md

```diff
@@ -0,0 +1,64 @@
+# Pack referenced projects
+
+- Author Name (https://github.com/maxkoshevoi)
+- Start Date (2022-03-28)
+- GitHub Issue (https://github.com/NuGet/Home/issues/3891)
+- GitHub PR (-)
+
+## Summary
+
+Allow referenced projects' artifacts to be packed inside the NuGet.
+
+## Motivation
+
+When SDK-style project is being packed as a NuGet, all projects that it references are considered as NuGet themselves that this NuGet depends on.
+This is not always a desired behavior, sometimes user needs artifacts from referenced project(s) to be packed into the NuGet.
+Currently there's no easy way to override this behavior except for ditching `csproj`, and  going back to `nuspec`, which becomes another source of truth and all that good stuff.
```

Comment:
```suggestion
Currently there's no easy way to override this behavior except for ditching `csproj`, and going back to `nuspec`. In many cases that involves piping lots of information from the `MSBuild` context into the `nuspec` evaluation via `key=value` parameters.  
Chaining together dozens of arguments to achieve parity with metadata usually automagically introduced by the SDK becomes tedious, which eventually weakens the otherwise rich set of information attached to a modern NuGet package.
```

Reply:
`<None Include="../otherproject/bin/$(Configuration)/x64/mydll.com" Pack="true" PackagePath="/runtimes/win-x64/native/" />` would do it right?

Reply:
Honestly people separate architectures by RuntimeIdentifiers nowadays.

Reply:
I’m not following? Runtime identifiers doesn’t really apply to nuget packages but to those consuming them. You first have to get those runtimes into the nuget

Reply:
@dotMorten What do you mean by "piping"? "Manually duplicating", or something else? Meaning of original wording was that we ending up storing the same information in two places (which complicates maintenance). Also the whole `nuspec` needs to be manfully created, even if most of the stuff in it can be inferred from `csproj`.

Reply:
I think the word piping was introduced by me in this convo so ill take this one - basically it is acknowledging that in an ideal world where people actually know about these properties you don't actually need to have an actual copy. 

For reference (one of my projects that would end up profiting from the overall feature):

https://github.com/MeikTranel/NXPorts/blob/master/src/NXPorts/NXPorts.csproj#L38-L48

Reply:
Cool, didn't know you could do that, thanks!